### PR TITLE
feat: label variant key UI

### PR DESCRIPTION
**Pull request by Zaimwa9**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have read the [Contributing Guide](/Flagsmith/flagsmith/blob/main/CONTRIBUTING.md).
- [ ] I have added information to `docs/` if required so people know about the feature.
- [x] I have filled in the "Changes" section below.
- [x] I have filled in the "How did you test this code" section below.

## Changes

Adds UI to set a **label** on multivariate variants (maps to the backend `key` field) and revamps the variants section of the feature modal's Value tab.

**Variant labels**

- Each variant shows an inline-editable label (slug characters only, max 255, `control` reserved, unique per feature).
- New variants get a default label (`Variant_n`, first free number) that is persisted on save; existing variants with no key fall back to a display-only placeholder.
- No save-flow changes — the key persists through the existing writable `/mv-options/` endpoint.

**Improvements**
- Added error management at the MV option level (returning invalid key or duplicated one)
- Error message appears in card
- Success response from MV options update only are also showing a success toast and updating the edition colorswatch

**Variants UI revamp**

- Each variant renders in a bordered, shadowed card: label header with a "Not saved" chip + delete icon.
- "Variants" section header with a primary "+ Add variant" button (outline "Create A/B/n Test" when none exist), and a tooltip explaining labels are required for experiments (behind `experimental_flags`).
- Control Value title shows the control percentage as a chip.
- `Input` gains reusable `underline` and `centered` props; aligned the value editor height with adjacent inputs.

**Experiment setup page**

- The variations table shows the variant label/key as the name and drops the unused Description column; `ContentCard` gains a `description` prop used by the setup step.

## How did you test this code?

- Extended the E2E flag test: `editVariantLabel` edits a variant's label via the new UI, saves, reopens the modal, and asserts persistence. Audited existing selectors (`add-variation`, `featureVariationValue*`, `featureVariationWeight*`, `delete-multivariate`) for compatibility with the new layout.
- Added Storybook stories for the new `Input` variants (`Underline`, `UnderlineCentered`).
- Manually, against a local API:
  1. Create a multivariate feature; add variants — they default to `Variant_1`, `Variant_2`…
  2. Rename a label; verify validation (spaces, `control`, duplicates rejected with inline errors) and the "Not saved" chip appearing only on edited variants.
  3. Save, reopen — labels persist; clear a label — it saves as `null` and the placeholder returns.

image
image

### CODE REVIEW

**Comment by Zaimwa9** [comment]
@gemini can you please have a thorough look of the data flow from the creation of a multivariate option, it's initial creation (POST) request and a direct update (PUT) without any full refresh

**Comment by Zaimwa9** [comment]
@gemini-code-assist review

**Review by talissoncosta — CHANGES_REQUESTED** [review]
Looks pretty @Zaimwa9 . 

Thanks for addressing the new design and writing the stories. 
I caught this on storybook:
1. The input underlined variant does not work well on dark mode yet. Do you mind double check that ?

Image

Overall looks great!

File: frontend/web/styles/components/_input.scss [hunk 1]
```diff
@@ -28,6 +28,56 @@ textarea {
 
 @import '~react-datepicker/dist/react-datepicker.css';
 
+// Borderless input with a bottom underline only; the underline turns
+// primary while focused. Used for inline edits (variant label, weight).
+.input-container.input-underline {
+  input.input {
+    border: none;
+    border-bottom: 1px solid $input-border-color;
```
Comment (talissoncosta) [inline frontend/web/styles/components/_input.scss:36]: Nit (non-blocking): these underline colours map cleanly onto our semantic tokens, which already flip per theme:

- `$input-border-color` / `$white-alpha-16` → `var(--color-border-default)`
- `$primary` (focus) → `var(--color-border-action)`
- hover `$basic-alpha-48` / `$white-alpha-48` → closest is `var(--color-border-strong)` (0.24, a touch subtler than 0.48)

Because the tokens are theme-aware, this would also let the separate `.dark .input-container.input-underline` block below collapse into the base rule.

No need to change anything here — we can fold this into #7758 (Migrate Input to TS), which tokenises the base input properly. Flagging just so the new CSS lines up with where we're heading. 🙏

File: frontend/web/components/modals/create-feature/tabs/FeatureValueTab.tsx [hunk 2]
```diff
@@ -155,13 +174,77 @@ const FeatureValueTab: FC<FeatureValueTabProps> = ({
...
+      'Failed to save this variation.'
+    )
+  })
+  const valueTitle = hasVariations ? (
+    <span className='d-inline-flex align-items-center'>
+      Control Value
+      <span className='ml-1'>
+        <Icon name='info-outlined' />
+      </span>
+      <span className='chip chip--xs ml-2'>{controlPercentage}%</span>
```
Comment (talissoncosta) [inline frontend/web/components/modals/create-feature/tabs/FeatureValueTab.tsx]: I think would be good if we standardise the Badge/Chip component. I see that this is a recurring pattern and we still don't have a proper component to handle that. 
No action to be done in here, just checking if you agree on that, so I can create an issue to handle that separately.

**Review by talissoncosta — DISMISSED** [review]

**Review by talissoncosta — DISMISSED** [review]

**Review by talissoncosta — APPROVED** [review]
